src/Ladybug_Shading Mask_II.py

```diff
@@ -294,27 +295,34 @@ def main(cenPt, context, radius, merge):
     # calculate the mask
     planarSrfs, crvsOnSky, skyDome, joinedContext = getSkyMask(cenPt, context, BBSky, BBRadius, merge)
     
-    
     # separate sky components
     maskedSkyDome = []
     unmaskedSkyDome = []
-    
+    zValue = []
+    brepList = []
     for faceCount in range(skyDome.Faces.Count):
         surface = skyDome.Faces.ExtractFace(faceCount)
         srfCenPt = rc.Geometry.AreaMassProperties.Compute(surface).Centroid
+        zValue.append(srfCenPt.Z)
         srfCenPt = rc.Geometry.Point3d(surface.ClosestPoint(srfCenPt))
         
         if not surface.IsValid:
             surface = surface.Faces.ExtractFace(0)
-            
-        if isMeshFaceVisible(cenPt, [srfCenPt], joinedContext):
-            unmaskedSkyDome.append(surface)
-        else:
-            maskedSkyDome.append(surface)
-    
-    # join!
-    #maskedSkyDome = list(rc.Geometry.Brep.JoinBreps(maskedSkyDome, sc.doc.ModelAbsoluteTolerance))
-    
+        # Making a list of breps in the skyDome
+        brepList.append(surface)
+        # Making a dictionary of brep : zValue of srfCenPt
+        brepDict = dict(zip(brepList, zValue))
+
+    # sorting the dictionary so that the brep with the largest zValue remains last in the
+    # dictionary
+    sortedDict = sorted(brepDict.items(), key = operator.itemgetter(1))
+    # The brep with smallest zValue for srfCenPt is assumed to be the unmaskedSkyDome
+    unmaskedSkyDome.append(sortedDict[0][0])
```

Comment:
@mostaphaRoudsari ,

The component was not outputting anything from it's maskedSkyDome output. When I looked, I found out that all the the breps were going to the unmaskedSkyDome list. Therefore, I came up with this idea to separate those breps.

If the user is using a regular shading geometry, It will not be too close to the cenPt. So the brep with SrfCenPt with highest Z value is going to be the the brep to be supplied to the maskedSkyDome.

Devang